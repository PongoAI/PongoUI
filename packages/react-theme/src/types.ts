import type { ColorSet } from '@pongo-ui/react-theme-generator';

export type Theme = ColorTokens & Fonts;

/**
 * Color theme tokens.
 */
export type ColorTokens = { textColor: string; canvasColor: string } & ColorSet<
  'inherit' | 'brand' | 'secondary' | 'success' | 'danger' | 'social' | 'warning' | 'info'
>;

export type Fonts = FontFamilyTokens & FontOpacityTokens & FontSizeTokens & FontLineHeightTokens & FontWeightTokens;

/**
 * Font family tokens.
 */
export type FontFamilyTokens = {
  baseFont: string;
  monospaceFont: string;
};

/**
 * Font opacity tokens
 */
export type FontOpacityTokens = {
  display: string;
  subHeadline: string;
  body: string;
  caption: string;
  title: string;
  header1: string;
  header2: string;
  header3: string;
};

/**
 * Font size tokens.
 */
export type FontSizeTokens = {
  fontSize100: string;
  fontSize200: string;
  fontSize300: string;
  fontSize400: string;
  fontSize500: string;
  fontSize600: string;
  fontSize700: string;
  fontSize800: string;
  fontSize900: string;
  fontSize1000: string;
};

/**
 * Font line height tokens.
 */
export type FontLineHeightTokens = {
  lineHeight100: string;
  lineHeight200: string;
  lineHeight300: string;
  lineHeight400: string;
  lineHeight500: string;
  lineHeight600: string;
  lineHeight700: string;
  lineHeight800: string;
  lineHeight900: string;
  lineHeight1000: string;
};

/**
 * Font weight tokens.
 */
export type FontWeightTokens = {
  thin: number;
  light: number;
  regular: number;
  medium: number;
  bold: number;
  black: number;
};
