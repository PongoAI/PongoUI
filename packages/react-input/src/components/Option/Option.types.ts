import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
import { MenuItem, MenuItemRadio } from '@pongo-ui/react-menu';

export type OptionSlots = {
  /**
   * The root of the Option.
   */
  root: Slot<typeof MenuItem & typeof MenuItemRadio>;
};

export type OptionCommons = {
  /**
   * The value of the option.
   */
  value: string;

  /**
   * Applies disabled styles to menu item but remains focusable
   */
  disabled?: boolean;
};

export type OptionProps = Omit<ComponentProps<OptionSlots>, 'name'> & OptionCommons;

export type OptionState = Omit<ComponentState<OptionSlots>, 'name'> & OptionCommons;
