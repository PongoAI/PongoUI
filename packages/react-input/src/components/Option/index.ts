export * from './Option';
export * from './Option.types';
export * from './Option';
export * from './useOption';
