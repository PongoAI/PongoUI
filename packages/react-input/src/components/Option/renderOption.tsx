import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { OptionState, OptionSlots } from './Option.types';

export const renderOption = (state: OptionState) => {
  const { slots, slotProps } = getSlots<OptionSlots>(state);

  return <slots.root {...slotProps.root} />;
};
