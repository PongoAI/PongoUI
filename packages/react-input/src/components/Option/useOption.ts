import * as React from 'react';
import { getNativeElementProps } from '@fluentui/react-utilities';
import type { OptionProps, OptionState } from './Option.types';
import { MenuItemRadio } from '@cebus/react-menu';

export const useOption = (props: OptionProps, ref: React.Ref<HTMLElement>): OptionState => {
  const { value, secondaryContent, icon, checkmark, disabled } = props;

  const state: OptionState = {
    value,
    components: {
      root: MenuItemRadio as any,
    },
    root: getNativeElementProps('div', {
      ref,
      ...props,
    }),
  };

  state.root.value = value;
  state.root.name = 'select';
  state.root.disabled = disabled;
  state.root.icon = icon;
  state.root.checkmark = checkmark;
  state.root.secondaryContent = secondaryContent;

  return state;
};
