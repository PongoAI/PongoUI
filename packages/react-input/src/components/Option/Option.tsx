import * as React from 'react';
import { renderOption } from './renderOption';
import { useOption } from './useOption';
import type { OptionProps } from './Option.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The option component used for children within Select.
 */
export const Option: ForwardRefComponent<OptionProps> = React.forwardRef((props, ref) => {
  const state = useOption(props, ref as any);

  return renderOption(state);
}) as ForwardRefComponent<OptionProps>;

Option.displayName = 'Option';
