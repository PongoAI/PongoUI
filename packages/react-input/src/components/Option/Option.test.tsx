import React from 'react';
import { Option } from './Option';
import { OptionProps } from './Option.types';
import { isConformant } from '../../common/isConformant';

describe('Option', () => {
  isConformant({
    Component: Option as React.FunctionComponent<OptionProps>,
    displayName: 'Option',
    disabledTests: ['make-styles-overrides-win'],
  });
});
