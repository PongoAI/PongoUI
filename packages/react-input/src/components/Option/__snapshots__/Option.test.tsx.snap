// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option renders a basic InputWrapper 1`] = `
<body>
  <div>
    <div
      aria-checked="false"
      class="___1x7e2fi f10pi13n f22iagw fvfw249 f1nbblvp f10sqwl7 f1ov4xf1 fne32i3 f8qjfyk fk6fouc f1arxnym f122n59 f1xqy1su fc467gg fnmxc1e fynvm4a f2hkw1w fdxsi5y f1i4q40k fa0joat f1yduw8h f1cwgisr f6dtr1x fvk4ntw f1bkt4b4 f16hf5f2 fyyqeim f1qhqcal f1kbdjx9 fw1d893 f14hlsw1 fpuz8dn fbw1f98 f8d6am2 f1gpnaou f11mktvj f1k6fduh fkcovh2 fc73jsq f8mbk1b f3c9x3h"
      name="select"
      role="menuitemradio"
      tabindex="0"
    >
      <span
        class="___lrcp890 fjw5fx7 fd461yt fd7fpy0"
      >
        <svg
          aria-hidden="true"
          class="___1yjlmar f1w7gpdv fez10in f1kxrtqx"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.05 3.49c.28.3.27.77-.04 1.06l-7.93 7.47A.85.85 0 014.9 12L2.22 9.28a.75.75 0 111.06-1.06l2.24 2.27 7.47-7.04a.75.75 0 011.06.04z"
          />
        </svg>
      </span>
    </div>
  </div>
</body>
`;
